### ISSUE TITLE
[Library] Discussion: Replace

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#381]
Add a function which has signature:

```
def replace[a](x: a, y: a, c: Coll[a]): Coll[a]
```

where `Coll` is Option, Result, List, Set or Map.

The semantics is that if the collection contains the given item `x` then it returns a new collection where that item has been replaced by `y`. Otherwise it returns the original collection.

In the case of a map, I think replace should take a pair `x: (k, v)` and `y: (k, v)`.

**Comment by magnus-madsen** [comment]
Liam wrote:

```
...  know you mentioned substitute or swap for this. What do you think about update? Scala has an almost identical function called updated.
```

**Comment by magnus-madsen** [comment]
Fixed in #407
